### PR TITLE
Refactor survey task choice questions styling

### PR DESCRIPTION
**Pull request by mcbouslog**
Package:
- lib-classifier

Describe your changes:
- create CheckBoxQuestion, CheckBoxOption, RadioQuestion, RadioOption components, tests, and (*Question) stories
- style options per [InVision](https://projects.invisionapp.com/d/main/default/#/console/12923997/444847259/inspect) and conversation with Becky
- refactor Questions component with CheckBoxQuestion and RadioQuestion
- cleanup stories
- move FilterButton component

Staging project and workflow:
- https://local.zooniverse.org:8080/?project=1634&workflow=2434

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxQuestion/CheckBoxQuestion.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { Box } from 'grommet'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import CheckBoxOption from './components/CheckBoxOption'
+
+export default function CheckBoxQuestion (props) {
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxQuestion/CheckBoxQuestion.js:7]: Probably been staring at too long, but maybe `CheckBoxQuestion` and `RadioQuestion` should be `CheckBoxOptions` and `RadioOptions` as individual question labels are rendered in `Questions` and really the components just represent the options?

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js [hunk 2]
```diff
@@ -771,19 +771,19 @@ const task = {
       'label': 'What behaviors do you see?',
       'answers': {
         'TNG': {
-          'label': ' Eating'
+          'label': 'Eating'
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js:774]: In option components, maybe should check if option label has a leading space, reformat to remove leading space, for backwards compatibility?

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxQuestion/CheckBoxQuestion.js (hunk 1 above)
Comment (srallen) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxQuestion/CheckBoxQuestion.js:7]: Or maybe `CheckBoxInputs` and `RadioInputs`? Not really required for this PR though.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js (hunk 2 above)
Comment (srallen) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js:774]: Trimming should work for this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim

**Review by srallen — COMMENTED** [review]
Without giving an explicit approval or request for changes, is there a plan for the hover and focus styles on the question inputs, both radio and checkbox? Additionally, there's this odd scroll to top behavior happening when I select something...

Did the discussion with Becky include dark theme styles? The look is a bit striking, imo, and it seems maybe we should consider similar styling to the image toolbar buttons because of their similarities (round circle buttons)? Here's what I mean:

Screen Shot 2021-05-03 at 3 42 47 PM

And the rectangle buttons maybe should use a border? We can chat a bit about it tomorrow morning.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js]: Now I'm not sure why I removed the leading whitespace to begin with, might have been before I refactored the labels with the Grommet Text component. After adding back leading space to mock data, before refactoring with `trim()`, the radio and checkbox labels appear ok (without leading space)?

**Comment by mcbouslog** [comment]
Per discussion, the following will be for subsequent PRs:
- dark theme styling
- focus and hover styling
- input borders
- scroll behavior

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/mock-data/task.js]: Screen Shot 2021-05-04 at 2 32 01 PM
Screen Shot 2021-05-04 at 2 31 55 PM

**Review by srallen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**